### PR TITLE
🐛 tilt: Enable deploy cert-manager per default again

### LINKED ISSUE #7100: Tilt no longer appears to install cert-manager and fails to start up
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]

Run `tilt up` on a clean kind cluster.

**What did you expect to happen:**

Error:

```
Build Failed: kubernetes apply: error mapping cert-manager.io/Certificate: no matches for kind "Certificate" in version "cert-manager.io/v1"
```

I have nothing else. It worked a couple commits ago...

This issue can be worked around if I run `clusterctl init --provider aws`. That installs cert-manager and things work from there on.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version: 1.2.1
- minikube/kind version: v0.11.1
- Kubernetes version: (use `kubectl version`): 1.22
- OS (e.g. from `/etc/os-release`): OSX

**Tilt setting:**

```json
{
    "enable_providers": [
      "kubeadm-bootstrap",
      "kubeadm-control-plane",
      "aws"
    ],
    "default_registry": "docker.io/skarlso/capa",
    "provider_repos": [
      "/Users/skarlso/goprojects/capa/cluster-api-provider-aws"
    ],
    "kustomize_substitutions": {
      "EXP_CLUSTER_RESOURCE_SET": "true",
      "EXP_MACHINE_POOL": "true",
      "AWS_B64ENCODED_CREDENTIALS": "cred==",
      "EXP_EKS_FARGATE": "false",
      "CAPA_EKS_IAM": "false",
      "CAPA_EKS_ADD_ROLES": "false",
      "EXP_BOOTSTRAP_FORMAT_IGNITION": "true"
    },
    "extra_args": {
      "aws": ["--v=2"]
    }
}

```

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7100

### CODE REVIEW

File: hack/tools/tilt-prepare/main.go [hunk 1]
```diff
@@ -161,12 +161,16 @@ func readTiltSettings(path string) (*tiltSettings, error) {
...
+	setDefaults(ts)
 	return ts, nil
 }
 
-// setDebugDefaults sets default values for debug related fields in tiltSettings.
-func setDebugDefaults(ts *tiltSettings) {
+// setDefaults sets default values for debug related fields in tiltSettings.
+func setDefaults(ts *tiltSettings) {
+	if ts.DeployCertManager == nil {
+		ts.DeployCertManager = pointer.BoolPtr(true)
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply hack/tools/tilt-prepare/main.go:171]: I know it's kind of duplicate with the nil check later, but I thought it's more straightforward to have an explicit defaulting + I wanted to keep the other place safe against nil pointers anyway :)

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Comment by sbueringer** [comment]
cc @ykakarap

**Review by ykakarap — APPROVED** [review]
/lgtm

**Review by chrischdi — COMMENTED** [review]
/lgtm

**Comment by chrischdi** [comment]
/retitle 🐛 tilt: Enable deploy cert-manager per default again

**Pull request merged**